include/aspect/introspection.h

```diff
@@ -294,6 +294,28 @@ namespace aspect
       IndexSets index_sets;
 
       /**
+       * A struct that describes the available methods for
+       * compositional fields. This type is used to determine how
+       * to advect a compositional field.
+       */
+      struct FieldMethod
+      {
+        enum kind
+        {
+          continuous_fem_field = 0,
+          discontinuous_fem_field = 1,
+          particles = 2
```

Comment:
I looked a bit into the issue and there are just too many places that are affected by this to include it in this PR. The option to have different finite elements for different compositional fields (including the one that are advected on particles) will have to wait for another PR.